flood_adapt/object_model/interface/object_model.py

```diff
@@ -0,0 +1,150 @@
+import logging
+import os
+from abc import ABC
+from pathlib import Path
+from typing import Any, Generic, Type, TypeVar
+
+import tomli
+import tomli_w
+from pydantic import BaseModel, Field
+
+from flood_adapt.misc.log import FloodAdaptLogging
+from flood_adapt.object_model.interface.path_builder import (
+    ObjectDir,
+)
+
+
+class IObjectModel(BaseModel):
+    """Base class for object models.
+
+    Attributes
+    ----------
+    name : str
+        Name of the object.
+    description : str (optional)
+        Description of the object.
+    """
+
+    name: str = Field(
+        ...,
+        description="Name of the object.",
+        min_length=1,
+        pattern='^[^<>:"/\\\\|?* ]*$',
```

Comment:
can you add a comment explaining what this pattern is supposed to catch? it's not very clear to me